tests/lib/rules/no-deprecated.js

```diff
@@ -48,9 +48,9 @@ ruleTester.run('no-deprecated', rule, {
     // Not deprecated
     'var element = React.createElement(\'p\', {}, null);',
     'var clone = React.cloneElement(element);',
-    'ReactDOM.render(element, container);',
-    'ReactDOM.unmountComponentAtNode(container);',
```

Comment:
we should be able to keep these test cases with the react version set to 17.999.999

Reply:
Added one more test to cover this scenario.

Reply:
I meant, keep the "valid" tests cases, because these remain valid in react < 18.

Reply:
I meant, keep the "valid" cases, because these remain valid in react < 18.

Reply:
You cannot leave it where it was before, React version isn't specified there.

Reply:
right - i mean, edit the existing rule to add the react version, so that "a passing test for ReactDOM.render" remains

Reply:
React version can be specified using eslint `settings.react.version` property which is demonstrated in the added tests. Introducing multiple sources to determine React version doesn't seem like a good idea to me.

Reply:
@ljharb Is there anything else that prevents the PR from being merged? I believe the requested scenario has been covered:
https://github.com/jsx-eslint/eslint-plugin-react/blob/1375bd1d893bdf73ee81e54bc73c5a317828d176/tests/lib/rules/no-deprecated.js#L122-L132